src/nom.rs

```diff
@@ -0,0 +1,201 @@
+    pub fn functions(&self) -> f64 {
+        self.functions.len() as f64
+    }
+
+    #[inline(always)]
+    pub fn closures(&self) -> f64 {
+        self.closures as f64
+    }
+
+    #[inline(always)]
+    pub fn total(&self) -> f64 {
+        self.functions() + self.closures()
+    }
+}
+
+pub trait Nom
+where
+    Self: Checker,
+{
+    fn compute(_node: &Node, _code: &[u8], _stats: &mut Stats) {}
+}
+
+impl Nom for PythonCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Python::*;
+
+        match node.kind_id().into() {
+            FunctionDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            Lambda | Lambda2 | Lambda3 => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for MozjsCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Mozjs::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for JavascriptCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Javascript::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for TypescriptCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Typescript::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for TsxCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Tsx::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for RustCode {
+    fn compute(node: &Node, code: &[u8], stats: &mut Stats) {
+        use Rust::*;
+
+        match node.kind_id().into() {
+            FunctionSignatureItem | FunctionItem => {
```

Comment:
If we have something like:
```rust
mod A { fn foo() {}}
mod B { fn foo() {}}
```
then when merging mod A (as a func space) and mod B in the unit then you'll have only one `foo` which is wrong